Release notes for MTV 2.6.3 (#526)
kubev2v/forklift-documentation: documentation/modules/rn-2.6.adoc

This part provides features and enhancements introduced in {project-full} 2.6:

.Preservation of static IPs in migrations from vSphere

You can now preserve static IPs of VMs that are installed with a Windows operating system and are migrated from vSphere. link:https://issues.redhat.com/browse/MTV-1148[(MTV-1148)]

.Support for migrating LUKS-encrypted devices in migrations from vSphere

> You can now migrate devices that have been encrypted by Linux Unified Key Setup (LUKS) when you create a vSphere migration plan. link:https://issues.redhat.com/browse/MTV-831[(MTV-831)]

Reviewer: (1) the decryption applies only to disks so I'd replace `devices` with `virtual disks`
(2) the user could select such VMs when creating a migration plan before so I'd drop the `when you create a vSphere migration plan` part
how about:
```suggestion
You can now migrate VMs with virtual disks that have been encrypted by Linux Unified Key Setup (LUKS) using cold migrations from a vSphere provider. link:https://issues.redhat.com/browse/MTV-831[(MTV-831)]
```

Author: Fixed, with mild rewrite.

.Specifying the primary disk when migrating from vSphere

> You can now specify the primary disk when you migrate VMs from vSphere which have more than one single bootable disk. This avoids {project-short} automatically attempting to convert the first bootable disk that it detects while it examines all the disks in a migration plan. This feature is needed because the first bootable disk is not necessarily the disk that the VM is expected to boot from in {virt}. link:https://issues.redhat.com/browse/MTV-1079[(MTV-1079)]

Reviewer: ```suggestion
You can now specify the primary disk when you migrate VMs from vSphere which have more than one bootable disk. This avoids {project-short} automatically attempting to convert the first bootable disk that it detects while it examines all the disks of a virtual machine. This feature is needed because the first bootable disk is not necessarily the disk that the VM is expected to boot from in {virt}. link:https://issues.redhat.com/browse/MTV-1079[(MTV-1079)]
```
(1) `one single` doesn't sound right, was it intentionally written this way?
(2) the examination is of the disks of a virtual machine (not of those in a migration plan)

.Providing a link to a remote provider UI

> You can now add a link to the UI of a remote cluster when you create a source provider. For example, if the provider is a remote {rhv-full} {rhv-short} cluster, you can add a link to the remote {rhv-short} web console when you define the provider. link:https://issues.redhat.com/browse/MTV-1054[(MTV-1054)]

Reviewer: @sgratch @yaacov  please check this part (is it something the user adds? I was under the impression that the link is auto-generated from the URL of the provider)

Reviewer 2: I think it's more correct to say:
"You now have a link to the  remote cluster management web based user interface"

reasons:
a. you do not add a link, it's auto generated.
b. you do not add it when you create the provider, you can edit it, in cases the auto generated list is not correct.

p.s.
we can add a few works on the reason for this link, a link to the provider management system makes it easy for users to manage and debug the migration on the source provider side with a direct link from MTV.

Author: @ahadas @yaacov @sgratch -- how about this? 

Links to remote provider UIs

You can now remotely access the UI of a remote cluster when you create a source provider. For example, if the provider is a remote {rhv-full} {rhv-short} cluster, {project-short} adds a link to the remote {rhv-short} web console when you define the provider. This feature makes it easier for you to manage and debug a migration from remote clusters.

Reviewer 2: sounds good to me

This part provides features and enhancements introduced in {project-full} 2.6:

.Preservation of a static IP in a migration from vSphere

> You can now preserve a static IP when importing VMs that have vSphere as their source provider. link:https://issuesredhat.com/browse/MTV-1148[(MTV-1148)]

Reviewer 3: Broken link 

```suggestion
You can now preserve a static IP when importing VMs that have vSphere as their source provider. link:https://issues.redhat.com/browse/MTV-1148[(MTV-1148)]
```

Reviewer: as it can be more than a single IP and it's limited to Windows, how about:
```suggestion
You can now preserve static IPs of VMs that are installed with Windows operating system and are migrated from vSphere. link:https://issues.redhat.com/browse/MTV-1148[(MTV-1148)]
```

> .Entering a list of LUKS decryption passphrases in a vSphere migration plan

Reviewer: this title and the description below make this look like it was possible to migrate LUKS-encrypted disks before by specifying a single decryption key and now it's possible specify multiple keys, while it wasn't possible to migrate VMs with LUKS-encrypted disks at all in previous versions - so this item needs to be rephrased

Edit: I think I see the source of this confusion - as the related Jira issues talk about changes in virt-v2v, however, we need to document the changes in MTV in this context

Author: Fixed.

> You can now enter a single list of multiple Linux Unified Key Setup (LUKS) decryption passphrases when you create a vSphere migration plan, rather than having to enter them one-by-one. For each LUKS-encrypted device, {project-short} tries each passphrase until one unlocks the device. link:https://issuesredhat.com/browse/MTV-831[(MTV-831)]

Reviewer 3: Broken link - 

```suggestion
You can now enter a single list of multiple Linux Unified Key Setup (LUKS) decryption passphrases when you create a vSphere migration plan, rather than having to enter them one-by-one. For each LUKS-encrypted device, {project-short} tries each passphrase until one unlocks the device. link:https://issues.redhat.com/browse/MTV-831[(MTV-831)]
```

.Specifying the primary disk when migrating from vSphere

> You can now specify the primary disk when you migrate VMs from vSphere which have both a primary and backup disk. This avoids {project-short} attempting to update the backup disk instead of the primary disk. link:https://issuesredhat.com/browse/MTV-1079[(MTV-1079)]

Reviewer 3: Broken link

```suggestion
You can now specify the primary disk when you migrate VMs from vSphere which have both a primary and backup disk. This avoids {project-short} attempting to update the backup disk instead of the primary disk. link:https://issues.redhat.com/browse/MTV-1079[(MTV-1079)]
```

> You can now specify the primary disk when you migrate VMs from vSphere which have both a primary and backup disk. This avoids {project-short} attempting to update the backup disk instead of the primary disk. link:https://issuesredhat.com/browse/MTV-1079[(MTV-1079)]

Reviewer: yeah, the term "primary" can be confusing here - it's not about primary vs. backup, but about which of the bootable disks needs to be considered as the disk that the VM would boot from in OpenShift (the conversion process makes adjustments only to a single operating system)
```suggestion
You can now specify the primary disk when you migrate VMs from vSphere which have more than a single bootable disk. This avoids {project-short} attempting to convert the first bootable disk that is detected while examining the disks that is not necessarily the disk that the VM is expected to boot from in OpenShift. link:https://issuesredhat.com/browse/MTV-1079[(MTV-1079)]
```

.Providing a link to a remote provider UI

> You can now add a link to the UI of a remote cluster when you create a source provider. For example, if the provider is a remote {rhv-full} {rhv-short} cluster, you can add a link to the remote {rhv-short} web console when you define the provider. link:https://issuesredhat.com/browse/MTV-1106[(MTV-1106)]

Reviewer 3: Broken link - 

```suggestion
You can now add a link to the UI of a remote cluster when you create a source provider. For example, if the provider is a remote {rhv-full} {rhv-short} cluster, you can add a link to the remote {rhv-short} web console when you define the provider. link:https://issues.redhat.com/browse/MTV-1106[(MTV-1106)]
```

.Migration from vSphere over a secure connection

> You can now specify a CA certificate that can be used to authenticate the server that runs vCenter or ESXi, depending on the specified SDK endpoint of the vSphere provider. link:https://issuesredhat.com/browse/MTV-530[(MTV-530)]

Reviewer 3: Broken link - 

```suggestion
You can now specify a CA certificate that can be used to authenticate the server that runs vCenter or ESXi, depending on the specified SDK endpoint of the vSphere provider. link:https://issues.redhat.com/browse/MTV-530[(MTV-530)]
```

For a complete list of all known issues in this release, see the list of link:https://issues.redhat.com/issues/?filter=12436209[Known Issues] in Jira.

[id="resolved-issues-26_{context}"]
== Resolved issues

This release has the following resolved issues:

.CVE-2024-24788: `golang: net` malformed DNS message can cause infinite loop

A flaw was discovered in the `stdlib` package of the Go programming language, which impacts previous versions of {project-short}. This vulnerability primarily threatens web-facing applications and services that rely on Go for DNS queries. This flaw has been resolved in {project-short} 2.6.3.

For more details, see link:https://access.redhat.com/security/cve/cve-2024-24788[(CVE-2024-24788)].

.Migration scheduling did not take into account that `virt-v2v` copies disks sequentially (vSphere only)

> In earlier releases of {project-short}, there was a problem with vSphere cold migrations to the {ocp-short} cluster that {project-short} runs on. In these migrations, {project-short} uses the `controller_max_vm_inflight` setting for vSphere to schedule migrations. In the past, the scheduling mechanism did not take into account that disks are transferred sequentially by `virt-v2v`, which led to `controller_max_vm_inflight` being exceeded and maigrations failing. This issue has been resolved in {project-short} 2.6.3. link:https://issues.redhat.com/browse/MTV-1191[(MTV-1191)]

Reviewer 3: You must use a spell check - you have typos `maigrations`

```suggestion
In earlier releases of {project-short}, there was a problem with vSphere cold migrations to the {ocp-short} cluster that {project-short} runs on. In these migrations, {project-short} uses the `controller_max_vm_inflight` setting for vSphere to schedule migrations. In the past, the scheduling mechanism did not take into account that disks are transferred sequentially by `virt-v2v`, which led to `controller_max_vm_inflight` being exceeded and migrations failing. This issue has been resolved in {project-short} 2.6.3. link:https://issues.redhat.com/browse/MTV-1191[(MTV-1191)]
```

Reviewer: > You must use a spell check - you have typos `maigrations`

actually that part should be replaced not just because of the typo but since migrations didn't fail due to this issue but rather that we didn't start as many migrations as we could have

.Cold migrations failed after changing the ESXi network (vSphere only)

> In earlier versions of {project-short}, cold migrations using an ESXi endpoint failed if any network was used except for the default network for data transfer. This issue has been resolved in {project-short} 2.6.3. link:https://issues.redhat.com/browse/MTV-1180[(MTV-1180)]

Reviewer: ```suggestion
In earlier versions of {project-short}, cold migrations from a vSphere provider with an ESXi SDK endpoint failed if any network was used except for the default network for disk transfers. This issue has been resolved in {project-short} 2.6.3. link:https://issues.redhat.com/browse/MTV-1180[(MTV-1180)]
```
(let's be more specific since the ESXi network is used only for disk transfers)

.Warm migrations over an ESXi network were stuck in `DiskTransfer` state (vSphere only)

> In earlier versions of {project-short}, warm migrations over an ESXi network using a vCenter endpoint were stuck in `DiskTransfer` state because {project-short} could not find the CDI importer pod for `DataVolume`. This caused {project-short} to be unable to locate the image snapshot. This issue has been resolved in {project-short} 2.6.3. link:https://issues.redhat.com/browse/MTV-1161[(MTV-1161)]

Reviewer: ```suggestion
In earlier versions of {project-short}, warm migrations over an ESXi network from a vSphere provider with a vCenter SDK endpoint were stuck in `DiskTransfer` state because {project-short} was unable to locate image snapshots. This issue has been resolved in {project-short} 2.6.3. link:https://issues.redhat.com/browse/MTV-1161[(MTV-1161)]
```
("{project-short} could not find the CDI importer pod for `DataVolume`" - that's a consequence of the inability to find the snapshot that we can drop)

.Leftover PVCs were in `Lost` state after cold migrations

> In earlier versions of {project-short}, after cold migrations, there were leftover PVCs that had a status of `Lost` instead of being deleted, even after the migration plan that contained them was archived and deleted. Investigation showed that this was because importer pods were retained after copying, by default, rather than in only specific cases. This issue has been resolved in {project-short} 2.6.3. link:https://issues.redhat.com/browse/MTV-1095[(MTV-1095)]

Reviewer: ```suggestion
In earlier versions of {project-short}, after cold migrations, there were leftover PVCs that had a status of `Lost` instead of being deleted, even after the migration plan that created them was archived and deleted. Investigation showed that this was because importer pods were retained after copying, by default, rather than in only specific cases. This issue has been resolved in {project-short} 2.6.3. link:https://issues.redhat.com/browse/MTV-1095[(MTV-1095)]
```

.Guest operating system from vSphere might be missing (vSphere only)

> In earlier versions of {project-short}, some VMs that were imported from vSphere were not mapped to a template in {ocp-short} while other VMs, with the same guest operating system, were mapped to the corresponding template. Investigation showed that this was because vSphere stops reporting the operating system after not getting updates from VMware tools for some time. This issue has been resolved in {project-short} 2.6.3 by taking the value of the operating system form the output of `virt-v2v` . link:https://issues.redhat.com/browse/MTV-1046[(MTV-1046)]

Reviewer 3: Mixed tenses used, and an extra space

```suggestion
In earlier versions of {project-short}, some VMs that were imported from vSphere were not mapped to a template in {ocp-short} while other VMs, with the same guest operating system, were mapped to the corresponding template. Investigation showed that this was because vSphere stopped reporting the operating system after not receiving updates from VMware tools for some time. This issue has been resolved in {project-short} 2.6.3 by taking the value of the operating system form the output of `virt-v2v`. link:https://issues.redhat.com/browse/MTV-1046[(MTV-1046)]
```

Reviewer: adding to @anarnold97 's proposed changes:
```suggestion
In earlier versions of {project-short}, some VMs that were imported from vSphere were not mapped to a template in {ocp-short} while other VMs, with the same guest operating system were mapped to the corresponding template. Investigations indicated that this was because vSphere stopped reporting the operating system after not receiving updates from VMware tools for some time. This issue has been resolved in {project-short} 2.6.3 by taking the value of the operating system the inspection of the disks by `virt-v2v`. link:https://issues.redhat.com/browse/MTV-1046[(MTV-1046)]
```
